### PR TITLE
Fixes #18115 - Logger is initialized when Syslog isn't available.

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by dmitri-d** [comment]
Another change is that smart-proxy exists now if syslog is configured but not available. In such a configuration ::Proxy::SETTINGS.log_file doesn't contain a usable path and default log file path may not be accessible to smart-proxy.

File: lib/proxy/log.rb [hunk 1]
```diff
@@ -32,43 +24,72 @@ def request_id
...
-        logger = ::Logger.new(STDOUT)
+        logger = stdout_logger
       elsif log_file.casecmp('SYSLOG').zero?
-        begin
-          logger = ::Syslog::Logger.new 'foreman-proxy'
-        rescue
-          logger = default_logger(log_file)
-          puts "'SYSLOG' logger is not supported on this platform, using file-based logger instead"
+        unless syslog_available?
+          puts "SYSLOG' logger is not supported on this platform, please use STDOUT or a file-based logger. Exiting..."
```
Comment (domcleal) [inline lib/proxy/log.rb]: Nitpick, typo introduced with a missing single quote before "SYSLOG".

**Review by domcleal — CHANGES_REQUESTED** [review]

**Comment by dmitri-d** [comment]
Fixed the typo and rebased.

**Review by domcleal — APPROVED** [review]

**Comment by domcleal** [comment]
Merged, thanks @witlessbird.

**Pull request merged**